explorer/testdata/let/value_expression_declaration.carbon

```diff
@@ -0,0 +1,79 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// CHECK:STDOUT: Setup
+// CHECK:STDOUT: 0: Heap{}, 1: C{.id = 1}, 2: C{.id = 2}
+// CHECK:STDOUT:
+// CHECK:STDOUT: Initialize c1 from reference expression
+// CHECK:STDOUT: 0: Heap{}, 1: C{.id = 1}, 2: C{.id = 2}, 3: C{.id = 1}
+// CHECK:STDOUT:
+// CHECK:STDOUT: Initialize c2 from value expression
+// CHECK:STDOUT: 0: Heap{}, 1: C{.id = 1}, 2: C{.id = 2}, 3: C{.id = 1}, 4: C{.id = 2}
```

Comment:
c1 is assigned from a `var`, and so a copy is useful in order to prevent observation of the `var` mutation (unless you get really technical, like observing `c1_var` doesn't change). However, `c2` is a `let`, and so it could reference `c2_let` without creating a copy. Does it match your expectation that that's the end goal, even if the code doesn't do it right now?

Reply:
I added a todo in `InitializePlaceholderValue`, and an action in the "next work" section of the PR. Let expressions for temporaries are not bound to allocations, causing the destructor to never be called. The safe option for now is to copy to ensure that the destructor is called at least once. So this isn't the end goal, yes.